### PR TITLE
Add Windows Images

### PR DESCRIPTION
**Pull request by ajschmidt8**
This PR adds images for Windows runners.

It includes the changes from #32 and therefore has that branch set as the base branch.

When this PR is ready to merge, I will close #32 and retarget this PR to `main` so that it includes the changes from both PRs.

**To Do:**

- [x] Add Docker mirror
- [x] Add runner start-up script
- [x] Add runner job initialization script
- [x] Fetch latest version of `gh` from GitHub releases
- [x] Fetch latest version of `docker` from releases page
- [x] Fetch latest version of `git` from GitHub releases
- [x] . Source powershell variables and environment vars in each installer.
- [x] Investigate whether we need `runner` user. If not, remove it.
- [x] Replace `win` with `windows` to match GitHub-hosted runner labels
- [x] Address remaining `TODOs`
- [x] Undo temporary changes to `matrix.yaml`

### CODE REVIEW

**Comment by wmaxey** [comment]
The final commit does a good bit of surgery to get things working. But on-prem builds work locally so I expect AWS to as well.

File: source.pkr.hcl [hunk 1]
```diff
@@ -76,7 +76,12 @@ source "amazon-ebs" "windows" {
 
   skip_create_ami   = !var.upload_ami
   shutdown_behavior = "terminate"
-  user_data_file    = "./win/init/bootstrap.ps1"
+  # the `user_data` file for AWS must be wrapped in a <powershell> tag
+  user_data    = <<-EOF
+  <powershell>
+  ${file("${path.root}/win/init/bootstrap.ps1")}
```
Comment (wmaxey) [inline source.pkr.hcl:82]: Wow

File: win/installers/runner.ps1 [hunk 2]
```diff
@@ -8,3 +11,11 @@ Invoke-WebRequest -UseBasicParsing -Uri "${actionRunnerUri}" -OutFile actions-ru
 # Extract the installer
 tar -xf actions-runner.zip
 Remove-Item actions-runner.zip
+
+# Setup runner scripts
+Copy-Item "${env:NV_CONTEXT_DIR}/initialize_runner.ps1" "C:/actions-runner/.initialize_runner.ps1"
+Copy-Item "${env:NV_CONTEXT_DIR}/runner.ps1" "C:/actions-runner/runner.ps1"
+
+if ($env:NV_COPY_JITCONFIG) {
```
Comment (wmaxey) [inline win/installers/runner.ps1:19]: This actually doesn't seem to work, I think that `false` from packer isn't quite the same as what powershell thinks as false. We may have to compare against a numeric or something.

File: build.pkr.hcl [hunk 3]
```diff
@@ -1,6 +1,6 @@
 build {
-  source "source.null.qemu_dependencies" {
-    name = "linux-premise"
+  source "source.null.preprovision" {
+    name = "linux-kvm"
```
[the remark this answers is not in the document]
Comment (ajschmidt8) [reply build.pkr.hcl]: fixed in [be408e7](https://github.com/nv-gha-runners/vm-images/pull/33/commits/be408e728f2b6ac30fb9db681b8cd9a6adcdbf5f)

File: source.pkr.hcl [hunk 4]
```diff
@@ -81,3 +68,95 @@ source "qemu" "ubuntu" {
...
+  net_device       = "virtio-net-pci"
+
+  # Disk space optimizations for detecting TRIM on Windows
+  disk_interface     = "virtio-scsi"
+  disk_discard       = "unmap"
+  disk_detect_zeroes = "unmap"
+  # Disk compression is *not* fast, maybe we consider enabling this inside the VM instead
+  # Currently compresses to about 6gb
+  #  disk_compression   = true
+  skip_compaction = false
```
Comment (ajschmidt8) [inline source.pkr.hcl:143]: We can defer all of this QEMU disk optimization stuff for later when we actually publish the QEMU Windows images

File: windows/installers/cloudbase-init.ps1 [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+Write-Output "cloudbase-init installation finished"
+
+Copy-Item "${env:NV_CONTEXT_DIR}/cloudbase-init.conf" "${env:ProgramFiles}/Cloudbase Solutions/Cloudbase-Init/conf/cloudbase-init.conf"
+
+Remove-Item CloudbaseInitSetup_Stable_x64.msi
+
+# Stop cloudbase to prevent reboot timeout
+Write-Output "Sleeping before stopping cloudbase to prevent reboot timeout"
+Start-Sleep -Seconds 5
+Stop-Service cloudbase-init
```
Comment (ajschmidt8) [inline windows/installers/cloudbase-init.ps1:17]: @wmaxey, could you elaborate on this? what is the reboot timeout? is that something specific cloudbase-init?

**Pull request merged**